dart/colfer_gen.dart

```diff
@@ -0,0 +1,831 @@
+      return false;
+    }
+    if (other.s_ != s_) {
+      print('compare prop s_ doesn\'t match $s_ != ${other.s_}');
+      return false;
+    }
+    if (!(IterableEquality().equals(other.a_, a_))) {
+      print('compare prop a_ doesn\'t match $a_ != ${other.a_}');
+      return false;
+    }
+    if (other.o != o) {
+      print('compare prop o doesn\'t match $o != ${other.o}');
+      return false;
+    }
+    if (!(IterableEquality().equals(other.os, os))) {
+      print('compare prop os doesn\'t match $os != ${other.os}');
+      return false;
+    }
+    if (!(IterableEquality().equals(other.ss, ss))) {
+      print('compare prop ss doesn\'t match $ss != ${other.ss}');
+      return false;
+    }
+    if (other.as_.length != as_.length) {
+      print('compare prop length of as_ doesn\'t match ${as_.length} != ${other.as_.length}');
+      return false;
+    }
+    for (int vi = 0; vi < as_.length; vi++) {
+      if (!(IterableEquality().equals(other.as_[vi], as_[vi]))) {
+        print('compare prop as_[$vi] doesn\'t match ${as_[vi]} != ${other.as_[vi]}');
+        return false;
+      }
+    }
+    if (other.u8 != u8) {
+      print('compare prop u8 doesn\'t match $u8 != ${other.u8}');
+      return false;
+    }
+    if (other.u16 != u16) {
+      print('compare prop u16 doesn\'t match $u16 != ${other.u16}');
+      return false;
+    }
+    if (!(IterableEquality().equals(other.f32s, f32s))) {
+      print('compare prop f32s doesn\'t match $f32s != ${other.f32s}');
+      return false;
+    }
+    if (!(IterableEquality().equals(other.f64s, f64s))) {
+      print('compare prop f64s doesn\'t match $f64s != ${other.f64s}');
+      return false;
+    }
+    return true;
+  }
+
+  @override
+  int get hashCode => hashObjects(
+      [b, u32, u64, i32, i64, f32, f64, t, s_, a_, o, os, ss, as_, u8, u16, f32s, f64s]);
+
+  O({
+    this.b = false,
+    this.u32 = 0,
+    this.u64 = 0,
+    this.i32 = 0,
+    this.i64 = 0,
+    this.f32 = 0.0,
+    this.f64 = 0.0,
+    this.t,
+    this.s_ = '',
+    Uint8List? a_,
+    this.o,
+    List<O>? os,
+    List<String>? ss,
+    List<Uint8List>? as_,
+    this.u8 = 0,
+    this.u16 = 0,
+    Float32List? f32s,
+    Float64List? f64s,
+  })  : a_ = a_ ?? Uint8List(0),
+        os = os ?? [],
+        ss = ss ?? [],
+        as_ = as_ ?? [],
+        f32s = f32s ?? Float32List(0),
+        f64s = f64s ?? Float64List(0);
+
+  /// Encodes as Colfer into buf and returns the number of bytes written.
+  /// It May throw RangeError or colfer related Exception.
+  int marshal(BytesBuilder buf) {
+    int i = buf.length;
+
+    if (b) {
+      buf.addByte(0);
+    }
+
+    if (u32 != 0) {
+      if (u32 > 4294967295 || u32 < 0) {
+        throw Exception('colfer: $u32 out of reach: u32');
+      }
+      if (u32 < 0x200000) {
+        buf.addByte(1);
+        encodeVarint(buf, u32);
+      } else {
+        buf.addByte(1 | 128);
+        buf.add(Uint8List(4)..buffer.asByteData().setInt32(0, u32));
```

Comment:
It looks like you allocate a new 4-byte array just to set for bytes in buf.

Reply:
This was the built in method I could find. BytesBuilder doesn't have a "set" method, only add. If we consider changing it, it would probably solve this.